Add v3 to v4 plugin migration page

dotNetkow opened the pull request.

dotNetkow commented on docs/main/updating/plugins/4-0.md:
```diff
@@ -0,0 +1,77 @@
+While you can follow the manual steps below, check out [this tool](https://github.com/rdlabo-team/capacitor-plugin-to-v4) from community member [@rdlabo](https://twitter.com/rdlabo) that automatically updates a Capacitor 3 plugin to v4.
+:::
+
+## Android
```
I followed [this commit](https://github.com/ionic-team/create-capacitor-plugin/commit/03027bf603626ebfdba084cb9d9a1a359d008554#diff-a875f0630e8a46d5a0e706a94502051f5a75e669312987096e4f52941b2bf476) to figure out what changed. I may have missed something. if it's easier, maybe we just recommend the migration tool?

jcesarmobile reviewed (changes requested):
Why are you removing the updating to 3.x guide? I think it can still be helpful and we are keeping the older upgrade guides for capacitor, so I think we should keep the old upgrade guides for plugins too.

You have missed the podspec change, which is the most important one related to the iOS deployment target as the podspec is what's used when users install the plugin in their project
https://github.com/ionic-team/create-capacitor-plugin/commit/03027bf603626ebfdba084cb9d9a1a359d008554#diff-dc8721a872ae0dcc9e412a0a0b13bc684f6b8de82abee362ce965cf6d216f2d4L14

dotNetkow commented:
> Why are you removing the updating to 3.x guide? I think it can still be helpful and we are keeping the older upgrade guides for capacitor, so I think we should keep the old upgrade guides for plugins too.

That's true, definitely should keep those around. The PR makes it look like I deleted it entirely but I only removed from the current v4 Capacitor docs root level. Essentially it's replaced with the new v4 page (docs/updating/plugins/4-0). The 3.x guide still lives under `/docs/v3/updating/plugins/3-0`.
 
> You have missed the podspec change, which is the most important one related to the iOS deployment target as the podspec is what's used when users install the plugin in their project [ionic-team/create-capacitor-plugin@03027bf#diff-dc8721a872ae0dcc9e412a0a0b13bc684f6b8de82abee362ce965cf6d216f2d4L14](https://github.com/ionic-team/create-capacitor-plugin/commit/03027bf603626ebfdba084cb9d9a1a359d008554#diff-dc8721a872ae0dcc9e412a0a0b13bc684f6b8de82abee362ce965cf6d216f2d4L14)

OK, updated the PR. see if it's correct now.

jcesarmobile commented on docs/main/updating/4-0.md:
````diff
@@ -41,6 +41,12 @@ post_install do |installer|
 end
 ```
 
+Additionally, update the `podspec` file at the root of the project:
````
You have put this in the app update guide, not in the plugin update guide

dotNetkow replied:
Oops! fixed

dotNetkow commented:
OK, added v3 back in. Here's the root (v4) docs now:
![image](https://user-images.githubusercontent.com/2545368/208174632-7bc6c8c3-0660-4281-8314-e3b25cc870dd.png)

jcesarmobile commented on docs/main/updating/plugins/4-0.md:
```diff
@@ -0,0 +1,79 @@
+
+### Update to Gradle 7
+
+Adjust your Gradle project settings in `File > Project Structure > Project`. The Android Gradle Plugin Version should be 7.2.1 or later and the Gradle Version should be 7.4.2 or later. Apply these changes and run a gradle sync by clicking on the Elephant Icon in the top right of Android Studio.
```
```suggestion
Adjust your Gradle project settings in `File > Project Structure > Project`. The Android Gradle Plugin Version should be 7.2.1 and the Gradle Version should be 7.4.2. Apply these changes and run a gradle sync by clicking on the Elephant Icon in the top right of Android Studio.
```
Better not use "or later" as newer versions could cause issues, stick to the ones we used in all plugins and Capacitor itself.

jcesarmobile commented on docs/main/updating/plugins/4-0.md:
```diff
@@ -0,0 +1,79 @@
+
+Do the following to your Xcode project: select the **Project** within the project editor and open the **Build Settings** tab. Under the **Deployment** section, change **iOS Deployment Target** to **iOS 13.0**. Repeat the same steps for any app **Targets**.
+
+Then, open `ios/App/Podfile` and update the iOS version to 13.0:
```
```suggestion
Then, open `ios/Podfile` and update the iOS version to 13.0:
```
The guide is about plugins, there is no App folder in plugins, the Podfile is inside the ios folder

jcesarmobile commented on docs/main/updating/plugins/4-0.md:
```diff
@@ -0,0 +1,79 @@
+
+### Update Android Project Variables
+
+In your `build.gradle` file, update your values to the following new minimums:
```
```suggestion
In your `android/build.gradle` file, update your values to the following new minimums:
```

jcesarmobile commented on docs/main/updating/plugins/4-0.md:
````diff
@@ -0,0 +1,79 @@
+
+In your `build.gradle` file, update your values to the following new minimums:
+
+```groovy
````
this block is confusing, looks like you copied the changes from the variables.gradle file, but in plugins there is no variables.gradle file and the values are in different places and with a different format (using `:` instead of `=` )
So maybe instead of a code block use a list saying
minSdkVersion to 22
compileSdkVersion to 32
etc.

jcesarmobile commented on docs/main/updating/plugins/4-0.md:
```diff
@@ -0,0 +1,79 @@
+
+### Remove `jcenter()` from the Gradle files
+
+In previous Capacitor versions, `jcenter()` was required due to our Cordova compatibility layer being hosted on Jcenter. However, we are now using the latest Cordova Android version, hosted on Maven Central. With this, you may be able to remove `jcenter()` entirely from your `build.gradle` file. If you are using other native dependencies, make sure they aren't hosted on Jcenter before removing it ...
```
```suggestion
In previous Capacitor versions, `jcenter()` was required due to our Cordova compatibility layer being hosted on Jcenter. However, we are now using the latest Cordova Android version, hosted on Maven Central. With this, you may be able to remove `jcenter()` entirely from your `android/build.gradle` file. If you are using other native dependencies, make sure they aren't hosted on Jcenter before removing it!
```

jcesarmobile reviewed (changes requested):
added a few more comments

also it's missing this java 11 change (some java 11 changes are mentioned, but not this) https://github.com/ionic-team/create-capacitor-plugin/commit/03027bf603626ebfdba084cb9d9a1a359d008554#diff-51795f26cbdfdde24931f6d0f9d6f047f4617fc6a6cb850029cd78c8e0e9b90dR39-R40
